gramps/gen/filters/rules/person/_hassoundexname.py

```diff
@@ -0,0 +1,92 @@
+#
+# Gramps - a GTK+/GNOME based genealogy program
+#
+# Copyright (C) 2002-2007  Donald N. Allingham
+# Copyright (C) 2007-2008  Brian G. Matherly
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+#-------------------------------------------------------------------------
+#
+# Standard Python modules
+#
+#-------------------------------------------------------------------------
+from ....const import GRAMPS_LOCALE as glocale
+_ = glocale.translation.sgettext
+
+#-------------------------------------------------------------------------
+#
+# Gramps modules
+#
+#-------------------------------------------------------------------------
+from .. import Rule
+from ....lib.nameorigintype import NameOriginType
+from gramps.gen.soundex import soundex
+
+#-------------------------------------------------------------------------
+#
+# HasNameOf
+#
+#-------------------------------------------------------------------------
+class HasSoundexName(Rule):
+    """Rule that checks for full or partial name matches"""
+
+    labels      =  [_('Name:')]
+    name        = _('Soundex match of People with the <name>')
+    description = _("Soundex Match of people with a specified name. First "
+                    "name, Surname, Call name, and Nickname are searched in "
+                    "primary and alternate names.")
+    category    = _('General filters')
+    allow_regex = False
+
+    def apply(self, db, person):
+        for name in [person.get_primary_name()] + person.get_alternate_names():
+            if self._match_name(name):
+                return True
+        return False
+
+    def _match_name(self, name):
+        if self.list[0] and self._soundex_match(0, name.get_first_name()):
```

Comment:
Actually, you always pass `param_index=0`, so why not just send `self.list[0]` directly to `_soundex_match`?